src/librustc_middle/ty/mod.rs

```diff
@@ -1615,12 +1615,33 @@ pub struct WithOptConstParam<T> {
 
 impl<T> WithOptConstParam<T> {
     /// Creates a new `WithOptConstParam` setting `const_param_did` to `None`.
+    #[inline(always)]
     pub fn unknown(did: T) -> WithOptConstParam<T> {
```

Comment:
I think the general guidelines is not inline for generic function.

Reply:
hm... this function is only instantiated for `WithOptConstParam<DefId>` and `WithOptConstParam<LocalDefId>`.

Would be kind of sad if this is a problem :/

Reply:
Wait, isn't this only for code in the standard library as inline causes that to get included in every CGU? I don't think that policy counts for compiler internals afaik :thinking: 

Might be misremembering something here though